src/modules/Dropdown/Dropdown.js

```diff
@@ -1056,6 +1092,29 @@ export default class Dropdown extends Component {
     }
   }
 
+  setOpenDirection = () => {
+    if (!this.ref) return
+
+    const { keepInViewPort } = this.props
+    if (!keepInViewPort) return
+
+    const menu = this.ref.querySelector('.menu.visible')
+
+    if (!menu) return
+
+    const dropdownRect = this.ref.getBoundingClientRect()
+    const itemsHeight = menu.clientHeight
+    const spaceAtTheBottom =
+      document.documentElement.clientHeight - dropdownRect.y - dropdownRect.height - itemsHeight
+    const spaceAtTheTop = dropdownRect.y - itemsHeight
+
+    if (spaceAtTheBottom < 0 && spaceAtTheTop > 0) {
+      this.setState({ shouldOpenUpward: true })
+    } else {
+      this.setState({ shouldOpenUpward: false })
+    }
+  }
```

Comment:
An if/else which results in true/false statements can be simplified to using the condition as the value:

```jsx
this.setState({ 
  shouldOpenUpward: spaceAtTheBottom < 0 && spaceAtTheTop > 0,
})
```

Reply:
In the case that both top and bottom are blocked, perhaps we should open to the side with the most available space.  I think this math is wrong, but something like:

```jsx
this.setState({ shouldOpenUpward: spaceAtTheBottom > spaceAtTheTop })
```